src/SonarQube.Client/Api/Requests/V5_10/GetIssuesRequest.cs

```diff
@@ -0,0 +1,96 @@
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using Newtonsoft.Json;
+using SonarQube.Client.Models;
+using SonarQube.Client.Messages;
+using SonarQube.Client.Services;
+
+namespace SonarQube.Client.Api.V5_10
+{
+    public class GetIssuesRequest : RequestBase<SonarQubeIssue[]>, IGetIssuesRequest
+    {
+        [JsonProperty("key")]
+        public virtual string ProjectKey { get; set; }
+
+        protected override string Path => "batch/issues";
+
+        protected async override Task<Result<SonarQubeIssue[]>> ReadResponseAsync(HttpResponseMessage httpResponse)
+        {
+            if (!httpResponse.IsSuccessStatusCode)
+            {
+                return new Result<SonarQubeIssue[]>(httpResponse, null);
+            }
+
+            var byteArray = await httpResponse.Content.ReadAsByteArrayAsync();
+            // Protobuf for C# throws when trying to read outside of the buffer and ReadAsStreamAsync returns a non
+            // seekable stream so we can't determine when to stop. The hack is to use an intermediate MemoryStream
+            // so we can control when to stop reading.
+            // Note we might want to use FileStream instead to avoid intensive memory usage.
+            using (var stream = new MemoryStream(byteArray))
+            {
+                return new Result<SonarQubeIssue[]>(httpResponse,
+                    ReadFromProtobufStream(stream, ServerIssue.Parser).Select(ToSonarQubeIssue).ToArray());
+            }
+        }
+
+        protected override SonarQubeIssue[] ParseResponse(string response)
+        {
+            throw new NotImplementedException("This method will not be called because we override ReadResponse.");
+        }
+
+        private static IEnumerable<T> ReadFromProtobufStream<T>(Stream stream, MessageParser<T> parser)
+            where T : IMessage<T>
+        {
+            while (stream.Position < stream.Length)
+            {
+                yield return parser.ParseDelimitedFrom(stream);
+            }
+        }
+
+        private static SonarQubeIssue ToSonarQubeIssue(ServerIssue issue) =>
+            new SonarQubeIssue(issue.Path, issue.Checksum, issue.Line, issue.Msg, issue.ModuleKey,
+                ParseResolutionState(issue.Resolution), issue.RuleKey);
+
+        public static SonarQubeIssueResolutionState ParseResolutionState(string resolution)
+        {
+            switch (resolution)
+            {
+                case "":
+                    return SonarQubeIssueResolutionState.Unresolved;
+                case "WONTFIX":
+                    return SonarQubeIssueResolutionState.WontFix;
+                case "FALSE-POSITIVE":
+                    return SonarQubeIssueResolutionState.FalsePositive;
+                case "FIXED":
+                    return SonarQubeIssueResolutionState.Fixed;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(resolution));
```

Comment:
It might be better to assert and ignore (or treat it as "unknown") if we get an unrecognised value.
It's not likely that a new resolution state would be added to SonarQube/Cloud, but if one is then all existing client installations using that version of SQ will start failing until we roll out an update and the customer installs it.

Reply:
I agree about this. However, asserting will happen only when debugging, e.g. when testing manually... Instead of asserting I would rather write a log entry, but since this exception was already there, I prefer to do it in another PR. WDYT?

Reply:
Ok: you might want to put a TODO in so we don't forget.

Reply:
#663